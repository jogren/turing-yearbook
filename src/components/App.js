import React, { Component } from 'react';
import Cohort from './Cohort';
import people from '../data/yearbook-data.js';
import './App.css';
import TuringForm from './TuringForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: people.staff,
      students: people.students
    }
  }

  addMember = (newMember, role) => {
    console.log(newMember)
    console.log(role)
    console.log(this.state[role])
    this.setState({ role: this.state[role].push(newMember) })
    // this.setState({ role: [...this.state[role], newMember] });
  }

  deleteMember = (id) => {
    let filteredStaff = this.state.staff.filter(staffMember => staffMember.id !== id);
    let filteredStudents = this.state.students.filter(student => student.id !== id);
    this.setState({ staff: filteredStaff, students: filteredStudents })
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <h1>Turing Yearbook</h1>
        </header>
        <TuringForm addMember={this.addMember}/>
        <h2>Staff</h2>
        <Cohort people={this.state.staff} deleteMember={this.deleteMember} />
        <h2>Student</h2>
        <Cohort people={this.state.students} deleteMember={this.deleteMember} />
      </div>
    );
  }
}

export default App;
