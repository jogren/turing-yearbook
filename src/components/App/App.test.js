import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state with a new member when addMember is called', () => {
    const wrapper = shallow(<App />);
    const newMember = { name: "Jacob Ogren", quote: "Do a good deed daily", superlative: "12345", id: 14, photo: "https://placekitten.com/200/300" };
    expect(wrapper.state('staff').length).toEqual(13)
    wrapper.instance().addMember(newMember, 'staff');
    expect(wrapper.state('staff').length).toEqual(14)
  });

  it('should delete a member and update state when deleteMember is called', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.state('staff').length).toEqual(14)
    wrapper.instance().deleteMember(1);
    expect(wrapper.state('staff').length).toEqual(13)
  });
})