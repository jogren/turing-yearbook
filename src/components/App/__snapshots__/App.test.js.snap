// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Turing Yearbook
    </h1>
  </header>
  <TuringForm
    addMember={[Function]}
  />
  <h2>
    Staff
  </h2>
  <Cohort
    deleteMember={[Function]}
    people={
      Array [
        Object {
          "id": 1,
          "name": "Eric",
          "photo": "./eric.jpg",
          "quote": "You miss all the shots you don't take.",
          "superlative": "Most Likely to Win the PGA",
        },
        Object {
          "id": 2,
          "name": "Amy",
          "photo": "./amy.jpg",
          "quote": "I believe in you, cuties!",
          "superlative": "Most Likely to Be an Actual Superhero",
        },
        Object {
          "id": 3,
          "name": "Khalid",
          "photo": "./khalid.png",
          "quote": "Don't put the world on your shoulders.",
          "superlative": "Most Likely to Save the Planet from Alien Invasion",
        },
        Object {
          "id": 4,
          "name": "Travis",
          "photo": "./travis.jpg",
          "quote": "You do you.",
          "superlative": "Most Likely to Win a Grammy",
        },
        Object {
          "id": 5,
          "name": "Leta",
          "photo": "./leta.jpg",
          "quote": "It's okay to be kind of a mess.",
          "superlative": "Most Likely to Go to the Moon",
        },
        Object {
          "id": 6,
          "name": "Christie",
          "photo": "./christie.jpg",
          "quote": "Every journey begins with a single step. So take the damn step!",
          "superlative": "Most Likely to Be President",
        },
        Object {
          "id": 7,
          "name": "Brittany",
          "photo": "./brittany.jpg",
          "quote": "It was Robbie.",
          "superlative": "Most Likely to Get Away with Murder",
        },
        Object {
          "id": 8,
          "name": "Robbie",
          "photo": "./robbie.jpg",
          "quote": "Cool.",
          "superlative": "Most Likely to Be Accused of Murder",
        },
        Object {
          "id": 9,
          "name": "Will",
          "photo": "./will.jpg",
          "quote": "Stop worrying and just do the thing.",
          "superlative": "Most Likely to Invent Cold Fusion",
        },
        Object {
          "id": 10,
          "name": "Pam",
          "photo": "./pam.jpg",
          "quote": "Shit!",
          "superlative": "Most Likely to Win a Gold Medal",
        },
        Object {
          "id": 11,
          "name": "David",
          "photo": "./david.JPG",
          "quote": "I love git.",
          "superlative": "Most Likely to Be Eaten by a Shark",
        },
        Object {
          "id": 12,
          "name": "Louisa",
          "photo": "./louisa.jpg",
          "quote": "YOLO",
          "superlative": "Most Likely to Win a Knife Fight",
        },
        Object {
          "id": 13,
          "name": "Jeff",
          "photo": "https://i.ytimg.com/vi/DhqzMc_LXgQ/maxresdefault.jpg",
          "quote": "::thoughtful nonsense::",
          "superlative": "Most Likely to Tell a Dad Joke",
        },
      ]
    }
  />
  <h2>
    Student
  </h2>
  <Cohort
    deleteMember={[Function]}
    people={
      Array [
        Object {
          "id": 21,
          "name": "Abigail",
          "photo": "https://placekitten.com/200/300",
          "quote": "In the end, it's not the years in your life that count. It's the life in your years.",
          "superlative": "",
        },
        Object {
          "id": 22,
          "name": "Brian",
          "photo": "https://placekitten.com/200/300",
          "quote": "The greatest day in your life and mine is when we take total responsibility for our attitudes. That's the day we truly grow up.",
          "superlative": "",
        },
        Object {
          "id": 23,
          "name": "Charlotte",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life is not a problem to be solved, but a reality to be experienced.",
          "superlative": "",
        },
        Object {
          "id": 24,
          "name": "Danesh",
          "photo": "https://placekitten.com/200/300",
          "quote": "What we think determines what happens to us, so if we want to change our lives, we need to stretch our minds.",
          "superlative": "",
        },
        Object {
          "id": 25,
          "name": "Evelyn",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life is ten percent what happens to you and ninety percent how you respond to it.",
          "superlative": "",
        },
        Object {
          "id": 26,
          "name": "Fatima",
          "photo": "https://placekitten.com/200/300",
          "quote": "Believe that life is worth living and your belief will help create the fact.",
          "superlative": "",
        },
        Object {
          "id": 27,
          "name": "Glen",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life isn't about finding yourself. Life is about creating yourself.",
          "superlative": "",
        },
        Object {
          "id": 28,
          "name": "Harriet",
          "photo": "https://placekitten.com/200/300",
          "quote": "There is more to life than increasing its speed.",
          "superlative": "",
        },
        Object {
          "id": 29,
          "name": "Isabella",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life is really simple, but we insist on making it complicated.",
          "superlative": "",
        },
        Object {
          "id": 210,
          "name": "Jorge",
          "photo": "https://placekitten.com/200/300",
          "quote": "Our prime purpose in this life is to help others. And if you can't help them, at least don't hurt them.",
          "superlative": "",
        },
        Object {
          "id": 211,
          "name": "Kelly",
          "photo": "https://placekitten.com/200/300",
          "quote": "There are three constants in life...change, choice and principles.",
          "superlative": "",
        },
        Object {
          "id": 212,
          "name": "Lucia",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life's most persistent and urgent question is, 'What are you doing for others?",
          "superlative": "",
        },
        Object {
          "id": 213,
          "name": "Mei",
          "photo": "https://placekitten.com/200/300",
          "quote": "Life is a series of natural and spontaneous changes. Don't resist them--that only creates sorrow.Let reality be reality.Let things flow naturally forward in whatever way they like.",
          "superlative": "",
        },
        Object {
          "id": 214,
          "name": "Nina",
          "photo": "https://placekitten.com/200/300",
          "quote": "Change is the law of life. And those who look only to the past or present are certain to miss the future.",
          "superlative": "",
        },
        Object {
          "id": 215,
          "name": "Ophus",
          "photo": "https://placekitten.com/200/300",
          "quote": "Only a life lived for others is a life worthwhile.",
          "superlative": "",
        },
        Object {
          "id": 216,
          "name": "Parker",
          "photo": "https://placekitten.com/200/300",
          "quote": "When life is too easy for us, we must beware or we may not be ready to meet the blows which sooner or later come to everyone, rich or poor.",
          "superlative": "",
        },
        Object {
          "id": 217,
          "name": "Quin",
          "photo": "https://placekitten.com/200/300",
          "quote": "God gave us the gift of life; it is up to us to give ourselves the gift of living well.",
          "superlative": "",
        },
        Object {
          "id": 218,
          "name": "Rosa",
          "photo": "https://placekitten.com/200/300",
          "quote": "We make a living by what we get, but we make a life by what we give.",
          "superlative": "",
        },
        Object {
          "id": 219,
          "name": "Sam",
          "photo": "https://placekitten.com/200/300",
          "quote": "All life is an experiment. The more experiments you make the better.",
          "superlative": "",
        },
        Object {
          "id": 220,
          "name": "Tyrese",
          "photo": "https://placekitten.com/200/300",
          "quote": "My mission in life is not merely to survive, but to thrive; and to do so with some passion, some compassion, some humor, and some style.",
          "superlative": "",
        },
        Object {
          "id": 221,
          "name": "Ursula",
          "photo": "https://placekitten.com/200/300",
          "quote": "Once you say you're going to settle for second, that's what happens to you in life.",
          "superlative": "",
        },
        Object {
          "id": 222,
          "name": "Vic",
          "photo": "https://placekitten.com/200/300",
          "quote": "There is no passion to be found playing small--in settling for a life that is less than the one you are capable of living.",
          "superlative": "",
        },
        Object {
          "id": 223,
          "name": "Whitney",
          "photo": "https://placekitten.com/200/300",
          "quote": "If you don't design your own life plan, chances are you'll fall into someone else's plan.And guess what they have planned for you ? Not much.",
          "superlative": "",
        },
        Object {
          "id": 224,
          "name": "Xavier",
          "photo": "https://placekitten.com/200/300",
          "quote": "I've failed over and over and over again in my life and that is why I succeed.",
          "superlative": "",
        },
        Object {
          "id": 225,
          "name": "Yolanda",
          "photo": "https://placekitten.com/200/300",
          "quote": "Many of life's failures are people who did not realize how close they were to success when they gave up.",
          "superlative": "",
        },
        Object {
          "id": 226,
          "name": "Zach",
          "photo": "https://placekitten.com/200/300",
          "quote": "People grow through experience if they meet life honestly and courageously. This is how character is built.",
          "superlative": "",
        },
      ]
    }
  />
</div>
`;
